lxd/firewall/drivers/drivers_xtables.go

```diff
@@ -1370,6 +1370,56 @@ func (d Xtables) InstanceClearRPFilter(projectName string, instanceName string,
 	return nil
 }
 
+// InstanceSetupNetPrio activates setting of skb->priority for the specified instance device on the host interface.
+func (d Xtables) InstanceSetupNetPrio(projectName string, instanceName string, deviceName string, netPrio int) error {
+	comment := fmt.Sprintf("%s netprio", d.instanceDeviceIPTablesComment(projectName, instanceName, deviceName))
+	class := fmt.Sprintf("%x:%x", uint16(uint32(netPrio)>>16), uint16(uint32(netPrio)&0xFFFF))
+	args := []string{
+		"-i", deviceName,
+		"-j", "CLASSIFY",
+		"--set-class", class,
+	}
+
+	// IPv4 filter.
+	err := d.iptablesPrepend(4, comment, "mangle", "FORWARD", args...)
+	if err != nil {
+		return err
+	}
+
+	// IPv6 filter if IPv6 is enabled.
+	if shared.PathExists("/proc/sys/net/ipv6") {
+		err = d.iptablesPrepend(6, comment, "mangle", "FORWARD", args...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// InstanceClearNetPrio removes setting of skb->priority for the specified instance device on the host interface.
+func (d Xtables) InstanceClearNetPrio(projectName string, instanceName string, deviceName string) error {
+	if deviceName == "" {
+		return fmt.Errorf("Failed clearing netprio rules for instance (%s): device name is empty", instanceName)
```

Comment:
I appreciate you've likely copied this from other places, but logging an instance name without a project name can be quite confusing if a LXD install has multiple projects containing the instances with the same name in multiple projects.

So really we should be doing something like this here:

```
return fmt.Errorf("Failed clearing netprio rules for instance %q in project %q: device name is empty", projectName, instanceName)
```
```

Reply:
Fixed. Thanks!